[seadragon2000341] iP

seadragon2000341 opened the pull request:
## **DukePro**
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

**And it is FREE!**

Features:
- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the <code style="color:red">main</code> method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

seadragon2000341 committed:
Level 1

seadragon2000341 committed:
Level 2

seadragon2000341 committed:
Level 3

seadragon2000341 committed:
Level 4

seadragon2000341 committed:
A-TextUiTesting

seadragon2000341 committed:
Level-5

seadragon2000341 committed:
Level-5

seadragon2000341 committed:
Level-6

seadragon2000341 committed:
Add Level 7

seadragon2000341 committed:
A-MoreOOP

seadragon2000341 committed:
Level-8

seadragon2000341 committed:
Merge branch 'branch-level-8'

seadragon2000341 committed:
A-Packages

seadragon2000341 committed:
Merge branch 'add-gradle-support'

seadragon2000341 committed:
Write javadocs

seadragon2000341 committed:
Code Standard

seadragon2000341 committed:
Add find functionality

seadragon2000341 committed:
Merge branch 'branch-A-JavaDoc'

seadragon2000341 committed:
Revised Coding Standards

seadragon2000341 committed:
Merge branch 'branch-Level-9'

seadragon2000341 committed:
Add gradle

seadragon2000341 committed:
Add junit test

seadragon2000341 committed:
Checkstyle

seadragon2000341 committed:
GUI

leonpoh reviewed (commented):
Looks good no major flaws

seadragon2000341 committed:
Storage.loadFile method does not check for valid text structure from
the file

Might throw error since the array might not be correct length.

Ensuring that the file we are loading have the correct syntax would
reduce the chance of runtime errors.

Let's include some assert checks to check for the validity of the task
syntax in the file

seadragon2000341 committed:
Check coding style

seadragon2000341 committed:
Merge pull request #1 from seadragon2000341/branch-A-Assertions

Storage.loadFile method does not check for valid text structure from

seadragon2000341 committed:
Merge branch 'master' into branch-A-CodeQuality

seadragon2000341 committed:
Merge pull request #2 from seadragon2000341/branch-A-CodeQuality

Check coding style

seadragon2000341 committed:
Level-10

seadragon2000341 committed:
Merge branch 'master' of https://github.com/seadragon2000341/ip

benjamin-wee commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    /**
+     * Gets desription of a todo task
+     * @param command full command
+     * @return description of todo task
+     */
```
I really like that your javadocs are informative and helps the reader understand your code! Keep up the good work!

benjamin-wee commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,71 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.FileWriter;
```
Perhaps you could sort the imports in alphabetical order according to the coding standards?

benjamin-wee commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,50 @@
+     * Method to read commands interactively
+     * @return command that is inputted by user
+     */
+    public String readCommand() {
+        System.out.println("Enter your command!");
+        return this.reader.nextLine();
+    }
+
+    /**
+     * Prints welcome message
+     */
+    public String showWelcome() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        return "Hello from\n" + logo + "Hello! I'm duke.Duke\nWhat can I do for you?";
+    }
+
+    /**
+     * Prints a goodbye message
+     */
+    public String showBye() {
```
Good job on following the naming conventions of the methods in this class. Keep up the good work!

benjamin-wee reviewed (commented):
I like that your code is very readable and easy to understand! Keep up the good work!

seadragon2000341 committed:
Added extension B-ViewSchedules

lennoxtr commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,33 @@
+package duke.task;
```
Your code is very easy to read. Keep up the good work!

lennoxtr commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+    /**
+     * View tasks on a specific date
+     */
+    public String view(String command) {
```
A more intuitive method name here might be more appropriate. I suggest changing it to viewOnDate
```suggestion
    public String viewOnDate(String command) {
```

lennoxtr commented on src/test/java/duke/task/DeadlineTest.java:
```diff
@@ -0,0 +1,19 @@
+package duke.task;
+import org.junit.jupiter.api.Test;
+
```
Should the imports be grouped together?

lennoxtr commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+
+    /**
+     * Remove a task
+     * @param i index of task in arraylist
```
it would be nice if you could add @return to the javadoc and explain what is being returned

lennoxtr commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+    /**
+     * Nicely prints out all the task
+     */
+    public String list() {
```
I think another name might be more appropriate here, since list is not very intuitive. 
```suggestion
    public String listAllTasks() {
```

lennoxtr reviewed (commented):
Overall, you have applied good OOP. Your checkstyle and coding standard is great too. Keep up the good work

seadragon2000341 committed:
Update README.md

seadragon2000341 committed:
Update README.md

seadragon2000341 committed:
Update README.md

seadragon2000341 committed:
Add Ui.PNG

seadragon2000341 committed:
Final checks

seadragon2000341 committed:
Update build.gradle
